app/models/user.py

```diff
@@ -68,6 +69,46 @@ class User(db.Model):
     created_at = db.Column(db.DateTime, default=datetime.now())
     deleted_at = db.Column(db.DateTime)
 
+    @hybrid_property
+    def password(self):
+        """
+        Hybrid property for password
+        :return:
+        """
+        return self._password
+
+    @password.setter
+    def password(self, password):
+        """
+        Setter for _password, saves hashed password, salt and reset_password string
+        :param password:
+        :return:
+        """
+        salt = generate_random_salt()
+        self._password = generate_password_hash(password, salt)
+        hash_ = random.getrandbits(128)
+        self.reset_password = str(hash_)
+        self.salt = salt
+
+    @hybrid_property
+    def email(self):
+        """
+        Hybrid property for email
+        :return:
+        """
+        return self._email
+
+    @email.setter
+    def email(self, email):
+        """
+        Setter for _email, can be only set once
+        :param email:
+        :return:
+        """
+        if self._email is None:
+            self._email = email
+        else:
+            raise AttributeError("Email cannot be modified")
 
```

Comment:
Travis is failing  due to the hybrid properties. Don't know the reason. Looking into it.